flink-test-utils-parent/flink-test-utils-junit/src/main/java/org/apache/flink/core/testutils/CheckedThread.java

```diff
@@ -120,7 +154,7 @@ private void checkError() throws Exception {
 
     private void checkFinished() throws Exception {
         if (getState() != State.TERMINATED) {
-            throw new Exception(
+            throw new TimeoutException(
```

Comment:
?

Reply:
I think maybe it's more reasonable to use TimeoutException here?